R/telemetry.R

```diff
@@ -721,7 +721,13 @@ Telemetry <- R6::R6Class( # nolint object_name_linter
       force_username = NULL
     ) {
       if (!is.null(force_username)) return(force_username)
-      if (is.null(session) || is.null(session$user)) return(NULL)
+      if (is.null(session) || is.null(session$user)) {
+        if (is.null(Sys.getenv("SHINYPROXY_USERNAME"))) {
+          return(NULL)
+        } else {
+          return(Sys.getenv("SHINYPROXY_USERNAME"))
+        }
+      }
       session$user
```

Comment:
The `Sys.getenv(...)` call will always return a string if that variable is not set.

``` r
Sys.getenv("NON_EXISTITING_VAR")
#> [1] ""
Sys.getenv("R_PLATFORM")
#> [1] "x86_64-pc-linux-gnu"

nzchar(Sys.getenv("R_PLATFORM"))
#> [1] TRUE
nzchar(Sys.getenv("SHINYPROXY_USERNAME"))
#> [1] FALSE
```

I suggest the following explicit call. WDYT?

```suggestion
      # Detect username from deployment
      if (isFALSE(is.null(session)) && isFALSE(is.null(session$user))) {
        session$user # POSIT Connect
      } else if (nzchar(Sys.getenv("SHINYPROXY_USERNAME")) {
        Sys.getenv("SHINYPROXY_USERNAME") # shinyproxy.io
      } else {
        NULL
      }
```

Reply:
That's right.
Sys.getenv("ANY_VARIABLE") will return "".

added 
if (is.null(session) || is.null(session$user)) {
        shinyproxy_username <- Sys.getenv("SHINYPROXY_USERNAME")
        if (shinyproxy_username == "") {
          return(NULL) # Return NULL if SHINYPROXY_USERNAME is not set (or empty)
        } else {
          return(shinyproxy_username)
        }
 }